src/overlays/actors/ovl_En_Tanron2/z_en_tanron2.c

```diff
@@ -69,54 +77,614 @@ static DamageTable D_80BB8170 = {
+            } else {
+                this->unk_15A--;
+            }
+        }
+
+        Collider_UpdateCylinder(&this->actor, &this->collider1);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider1.base);
+    }
+
+    if (!(this->unk_148 & 0x1F)) {
+        if ((this->actionFunc != func_80BB69FC) && (this->actor.xyzDistToPlayerSq < 40000.0f)) {
+            this->unk_15B = 1;
+        } else {
+            this->unk_15B = 0;
+        }
+
+        if (this->actionFunc == func_80BB69FC) {
+            s16 atan = Math_Atan2S(this->actor.world.pos.x - D_80BB8450->unk_6BC.x,
+                                   this->actor.world.pos.z - D_80BB8450->unk_6BC.z);
+
+            if (ABS_ALT(BINANG_SUB(D_80BB8450->actor.yawTowardsPlayer, atan)) > 0x3000) {
+                this->unk_159 = 0;
+                this->actor.flags &= ~ACTOR_FLAG_1;
+            } else {
+                this->unk_159 = 1;
+                this->actor.flags |= ACTOR_FLAG_1;
+            }
+        }
+    }
+
+    if (CHECK_FLAG_ALL(this->actor.flags, ACTOR_FLAG_2000) && (this->actor.xzDistToPlayer < 80.0f)) {
+        this->actor.flags &= ~ACTOR_FLAG_2000;
+        this->unk_15A = 25;
+        this->unk_159 = 1;
+    }
+
+    input = CONTROLLER3(globalCtx);
+    if (CHECK_BTN_ALL(input->press.button, BTN_L)) {
+        this->unk_158 = 1;
+    }
+
+    if (CHECK_BTN_ALL(input->press.button, BTN_R)) {
+        func_80BB6B80(this);
+    }
+}
+
+void func_80BB7B90(Actor* thisx, GlobalContext* globalCtx) {
+    EnTanron2* this = THIS;
+
+    D_80BB8454 = (Math_SinS(globalCtx->gameplayFrames * 0x3000) * 0.1f) + 1.0f;
+    if (D_80BB8450->unk_1F6 == 11) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+#ifdef NON_MATCHING
+void EnTanron2_Draw(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnTanron2* this = THIS;
+    s32 i;
+    s32 j;
+    s32 found;
+    Actor* tanron2;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C2DC(globalCtx->state.gfxCtx);
+
+    gSPDisplayList(POLY_XLU_DISP++, object_boss04_DL_003450);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, 150);
+
+    tanron2 = globalCtx->actorCtx.actorLists[ACTORCAT_BOSS].first;
+
+    for (i = 0; i < ARRAY_COUNT(D_80BB8458); i++) {
+        D_80BB8458[i] = NULL;
+    }
+
+    found = 0;
+    while (tanron2 != NULL) {
+        if (tanron2->params < 100) {
+            D_80BB8458[found] = (EnTanron2*)tanron2;
+            found++;
+        }
+        tanron2 = tanron2->next;
+    }
+
+    for (j = 0; j < found - 1; j++) {
+        for (i = 0; i < found - 1; i++) {
+            if (D_80BB8458[i + 1] != NULL) {
+                if (D_80BB8458[i]->actor.projectedPos.z < D_80BB8458[i + 1]->actor.projectedPos.z) {
+                    EnTanron2* curr = D_80BB8458[i];
+
+                    D_80BB8458[i] = D_80BB8458[i + 1];
+                    D_80BB8458[i + 1] = curr;
```

Comment:
```suggestion
                    SWAP(EnTanron2*, D_80BB8458[i], D_80BB8458[i+1]);
```
Could use the SWAP macro here